### DISCUSSION TITLE
Apiman ver.x debug

### DISCUSSION THREAD

**Discussion by potehvad** [apiman/apiman#1503] [Help] [answered]
I am running apiman in docker container on ver.x launched by
`
java -jar /srv/gateway/apiman-distro-vertx-2.1.0.Final  --conf conf.json"
`

How can I attach to gateway with debbuger. Setting up JAVA_DEBUG=true and JAVA_DEBUG_PORT does not help me.

**Comment by msavy** [comment, marked as answer]
Here's the argument that you should pass in during 'run'.

Set $DEBUG_PORT to be whatever you prefer

-agentlib:jdwp=transport=dt_socket,address=0.0.0.0:$DEBUG_PORT,server=y,suspend=n

**Reply by msavy** [reply]
Did it work? If so, can you please mark the answer as accepted

**Reply by potehvad** [reply]
My problem was java version 8. Now I updated it to java 11. Debug works now.
